src/main/scala/com/fulcrumgenomics/alignment/Alignment.scala

```diff
@@ -284,8 +274,22 @@ case class Alignment(query: Array[Byte],
           targetBuffer.append(this.target(tOffset).toChar)
           tOffset += 1
         }
-      case other =>
-        throw new IllegalStateException(s"Padding string cannot support cigar operator $other")
+      case CigarElem(op, len) =>
+        forloop (from=0, until=len) { i =>
+          val q = this.query(qOffset).toChar
+          val t = this.target(tOffset).toChar
+          queryBuffer.append(q)
+          targetBuffer.append(t)
+          op match {
+            case CigarOperator.EQ => alignBuffer.append(matchChar)
+            case CigarOperator.X  => alignBuffer.append(mismatchChar)
+            case CigarOperator.M  => alignBuffer.append(if (q == t) matchChar else mismatchChar)
+            case other            => throw new IllegalStateException(s"Unsupported cigar operator: $other")
+          }
+
+          qOffset += 1
+          tOffset += 1
+        }
```

Comment:
why was this moved to the last `case` statement?  I would think it happens the most frequently, so it would be best to have it first?

Reply:
I had to either move it to the end, or turn it into three cases, because I want it to match all three of `=/X/M`, but also to know which one it was.  I've converted the code to match just on the operator so it should be more efficient, to make up for the change.